src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -230,25 +229,23 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleExit();
             }
 
-            if (commandResult.getPanel() != "") {
-                String newPanel = commandResult.getPanel();
-                if (!newPanel.equals(activePanel)) {
-                    hidePanels();
-                    switch (newPanel) {
-                    case "TASKS":
-                        setTaskListPanel();
-                        break;
-                    case "STUDENTS":
-                        setPersonListPanel();
-                        break;
-                    default:
-                        System.out.println("unknown panel asked for");
-                        break;
-                    }
-
+            if (!commandResult.getState().equals("")) {
```

Comment:
For this part, would the state be an empty "" only when no list commands have been input yet?

Reply:
it would be ""as for all non list commands, since they do not alter the state.
refer to command result constructors

Reply:
Alright noted, Thanks!